lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2639,10 +2625,18 @@ public function createEntity($className, array $data, &$hints = array())
 
                         if ($joinColumnValue !== null) {
                             if ($targetClass->containsForeignIdentifier) {
+                                if ($joinColumnValue instanceof AssociationCacheEntry) {
```

Comment:
@Ocramius It was my mistake... fixed with https://github.com/goetas/doctrine2/commit/268426aa85e56a7bf487dd706e81ba24764b8393
/cc @FabioBatSilva